### ISSUE TITLE
Pairs plot of model-implied trait values

### ISSUE THREAD

**Issue by ScottClaessens** [ScottClaessens/coevolve#54]
Currently, users are left somewhat blind as to the model-implied trait values for taxa. This is important because this can help users better understand the trait spaces in the flowfield and selection gradient plots.

To this end, it would be good to produce a pairs plot with scatterplots that visualise the median posterior trait values as points. Users could plot for all variables or a subset. Individual plots would look like this plot in the pre-print:
![image](https://github.com/user-attachments/assets/2bbc0af5-5b98-4d4a-8905-54e74141898f)

**Comment by ScottClaessens** [comment]
Or they could even just be histograms of trait values rather than pairs plots?

**Comment by ErikRingen** [comment]
> Or they could even just be histograms of trait values rather than pairs plots?

Both would be nice. And to incorporate uncertainty, at least as an optional argument. With the scatterplots this can be done with errorbars. With density plots for the marginal distributions we can use overlapping density plots just like pp_check() does.

**Comment by ScottClaessens** [comment]
Or we could use ellipses to visualise uncertainty, as Richard McElreath often does in his book, e.g.

![image](https://github.com/user-attachments/assets/1415c9db-bf48-4e22-8f05-3856d0a5a723)

**Comment by ErikRingen** [comment]
Ellipses look good, but I am not sure how well they scale when the number of data points increases. 

I like the look of this function from the python library BayesPlot, which we might use as inspiration:

<img width="660" alt="image" src="https://github.com/user-attachments/assets/6d660504-2e50-49ad-ba64-44f959763c08">

I would replace the diagonal histograms with pp_check() style density plots to show uncertainty in the marginal distributions.

**Comment by ScottClaessens** [comment]
Good idea! The plot looks nice. Busy today, but I'll implement this soon.

**Comment by ScottClaessens** [comment]
Okay, I've pushed a draft version of the function `coev_plot_trait_values()`. It was a little tricky to wrangle this plot: ggplot2 doesn't like having axes that do not match the data, making the upper row difficult to line up. I opted to remove the y-axis labels from the top left plot, moving away from how `ggpairs()` is implemented (see [here](https://ggobi.github.io/ggally/reference/ggpairs.html#examples)).

I think the end result looks quite nice:

```r
library(coevolve)

fit <-
  coev_fit(
    data = authority$data,
    variables = list(
      political_authority = "ordered_logistic",
      religious_authority = "ordered_logistic"
    ),
    id = "language",
    tree = authority$phylogeny,
    # manually set prior
    prior = list(A_offdiag = "normal(0, 2)"),
    # arguments for cmdstanr
    parallel_chains = 4,
    iter_sampling = 400,
    seed = 1
  )

coev_plot_trait_values(fit, xlim = c(-5, 7), ylim = c(-5, 7))
```

![image](https://github.com/user-attachments/assets/841e26ed-6aa9-4e25-8cd9-1e4a3ea540ea)

I'm yet to implement unit tests, etc. But @ErikRingen let me know what you think.

**Comment by ErikRingen** [comment]
This looks great! Does the user need to define the range of trait values or can it be inferred automatically based on the posterior draws?

**Comment by ScottClaessens** [comment]
It's not necessary to define the limits. I've set it so that the plot uses the min/max median trait values as limits. That does mean though that that the heatmap tends to get awkwardly cut off in the default plots. Open to other suggestions!

**Comment by ScottClaessens** [comment]
I've implemented unit tests in 31dd557. Closing this now, but feel free to re-open if we want to tweak these plots.
